devicetypes/fibargroup/fibaro-walli-dimmer-switch.src/fibaro-walli-dimmer-switch.groovy

```diff
@@ -0,0 +1,525 @@
+	log.debug "refresh()"
+
+	encapSequence([
+			zwave.switchMultilevelV1.switchMultilevelGet(),
+			meterGet(scale: 0),
+			meterGet(scale: 2),
+	], 1000)
+}
+
+def reset() {
+	encapSequence([
+			meterReset(),
+			meterGet(scale: 0)
+	])
+}
+
+def meterGet(scale) {
+	zwave.meterV2.meterGet(scale)
+}
+
+def meterReset() {
+	zwave.meterV2.meterReset()
+}
+
+private encapSequence(cmds, Integer delay=250) {
+	delayBetween(cmds.collect{ encap(it) }, delay)
+}
+
+private encap(physicalgraph.zwave.Command cmd) {
+	if (zwaveInfo?.zw?.contains("s")) {
+		secEncap(cmd)
+	} else {
+		log.debug "no encapsulation supported for command: $cmd"
+		cmd.format()
+	}
+}
+
+private secEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using Secure Encapsulation, command: $cmd"
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private getParameterMap() {[
+	[
+		name: "LED frame – colour when ON", key: "ledFrame–ColourWhenOn", type: "enum",
+		parameterNumber: 11, size: 1, defaultValue: 1,
+		values: [
+			0: "LED disabled",
+			1: "White",
+			2: "Red",
+			3: "Green",
+			4: "Blue",
+			5: "Yellow",
+			6: "Cyan",
+			7: "Magenta",
+			8: "colour changes smoothly depending on the measured power",
+			9: "colour changes in steps depending on the measured power"
+		],
+		description: "LED colour when the device is ON. When set to 8 or 9, LED frame colour will change depending on the measured power and parameter 10. Other colours are set permanently and do not depend on the power consumption."
+	],
+	[
+		name: "LED frame – colour when OFF", key: "ledFrame–ColourWhenOff", type: "enum",
+		parameterNumber: 12, size: 1, defaultValue: 0,
+		values: [
+			0: "LED disabled",
+			1: "White",
+			2: "Red",
+			3: "Green",
+			4: "Blue",
+			5: "Yellow",
+			6: "Cyan",
+			7: "Magenta"
+		],
+		description: "LED colour when the device is OFF."
+	],
+	[
+		name: "LED frame – brightness", key: "ledFrame–Brightness", type: "range",
+		parameterNumber: 13, size: 1, defaultValue: 100,
+		range: "0..102",
+		description: "Adjust the LED frame brightness. " +
+				"101 - brightness directly proportional to set level, " +
+				"102 - brightness inversely proportional to set level"
+	],
+	[
+		name: "Manual control – dimming step size", key: "manualControl–DimmingStepSize", type: "range",
+		parameterNumber: 156, size: 1, defaultValue: 1,
+		range: "1..99",
+		description: "Percentage value of the dimming step during the manual control. " +
+				"1 - [MSB] – Notification Type (Default)\n" +
+				"2 - Notification Value\n" +
+				"3 - Event/State Parameters\n" +
+				"4 - [LSB] – action\n" +
+				"1 to 99 - (1-99%, 1% step) – dimming step"
+	],
+	[
+		name: "Manual control – time of dimming step", key: "manualControl–TimeOfDimmingStep", type: "range",
+		parameterNumber: 157, size: 2, defaultValue: 5,
+		range: "0..255",
+		description: "Time of performing a single dimming step set in parameter 156 during the manual control."
```

Comment:
Time **to perform** ... **(Set in dimming step size parameter)**

I don't think the parameter numbers are visible.

Reply:
ok, fixed